lib/services/amazonka-dynamodb/src/Amazonka/DynamoDB/Internal.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StrictData #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wall -Werror #-}
+
+-- |
+-- Module      : Amazonka.S3.Internal
+-- Copyright   : (c) 2013-2021 Brendan Hay
+-- License     : Mozilla Public License, v. 2.0.
+-- Maintainer  : Brendan Hay <brendan.g.hay@gmail.com>
+-- Stability   : experimental
+-- Portability : non-portable (GHC extensions)
+module Amazonka.DynamoDB.Internal
+  ( -- * Attribute Values
+    AttributeValue (..),
+
+    -- ** Attribute Value Prisms
+    _L,
+    _NS,
+    _M,
+    _NULL,
+    _N,
+    _BS,
+    _B,
+    _SS,
+    _S,
+    _BOOL,
+  )
+where
+
+import Amazonka.Core
+import Amazonka.Prelude
+import Control.Lens.TH (makePrisms)
+import Data.Aeson
+
+#if MIN_VERSION_aeson(2,0,0)
+import qualified  Data.Aeson.KeyMap as KeyMap
+#else
+import qualified  Data.HashMap.Strict as KeyMap
+#endif
+
+-- | Represents the data for an attribute.
+--
+-- DynamoDB sends and receives JSON objects which contain a single
+-- item whose key is a data type and the value is the data itself. We
+-- provide an actual sum type to interact with these.
+--
+-- For more information, see
+-- <https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes Data Types>
+-- in the /Amazon DynamoDB Developer Guide/.
+data AttributeValue
+  = -- | An attribute of type List. For example:
+    --
+    -- @\"L\": [ {\"S\": \"Cookies\"} , {\"S\": \"Coffee\"}, {\"N\", \"3.14159\"}]@
+    L [AttributeValue]
```

Comment:
Is there a good reason for this to be a list and not a vector? I assume it comes out of the aeson `Value` as a `Vector`

Reply:
Not to my knowledge, I just reached for the thing that took the least typing.

Reply:
aeson has gone through the hard work of figuring out the length of this and giving us constant time access, I'd hate to lose that or have to re-do that. Anyone can use toList if they want to.

Reply:
OK